data/sea/06-segv.h

```diff
@@ -0,0 +1,44 @@
+#include "05-error.h"
+
+/* this global is pretty nasty, don't know how else we can get
+ * information in a signal handler though */
+#define SEGV_USER_DATA_SIZE (1024*1024)
+static char segv_user_data[SEGV_USER_DATA_SIZE];
+
+static void segv_handler (int sig)
+{
+    void *frames[50];
+    size_t n_frames = backtrace (frames, sizeof(frames));
+
+    fprintf (stderr, "icicle: segmentation fault:\n");
+    backtrace_symbols_fd (frames, n_frames, STDERR_FILENO);
+    fprintf (stderr, "\n%s\n", segv_user_data);
```

Comment:
I think i'll change this to put the default one back as the first thing it does, then hope that doing that doesn't segfault.